src/Tasks/AssemblyDependency/TaskExecutionContextExtension.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Tasks
+{
+    public static class TaskExecutionContextExtension
+    {
+        /// <summary>
+        /// Absolutize the given path with the startup directory.
+        /// </summary>
+        /// <param name="taskExecutionContext"></param>
+        /// <param name="path">Relative or absolute path.</param>
+        /// <returns></returns>
+        public static string GetFullPath(this TaskExecutionContext taskExecutionContext,  string path)
+        {
+            if (String.IsNullOrEmpty(taskExecutionContext.StartupDirectory) || String.IsNullOrEmpty(path))
```

Comment:
This is almost identical to FileUtilities.GetFullPath, though that does some extra escaping/unescaping. Do you think any of that is necessary here, and if so, should we just call that?

Reply:
FileUtilities.GetFullPath gives absolute path using the wrong current directory, without virtualization. Here we use taskExecutionContext.StartupDirectory as our current directory to complete the relative path (if it exists).

Reply:
FileUtilities has an overload to let you specify currentDirectory:
https://github.com/dotnet/msbuild/blob/369631b4b21ef485f4d6f35e16b0c839a971b0e9/src/Shared/FileUtilities.cs#L661-L680